ami_assets/snippets/iframe-auto-height.html

```diff
@@ -0,0 +1,35 @@
+<script id="iframe-auto-height" type="module">
+const iframe = document.getElementsByTagName('iframe')[0];
+const isIframeLoaded = (iframe.contentWindow.document.readyState === 'complete');
+
+function resizeIframe(iframe) {
+  const scrollHeight = iframe.contentWindow.document.documentElement.scrollHeight;
+  const bufferToAvoidScrollbar = 1;
+  iframe.style.height = scrollHeight + bufferToAvoidScrollbar + 'px';
+}
+
+/* to avoid both flicker effect and excessive calls when resizing iframe */
+function debounce(fn, delay = 50) {
+  let timeoutId;
+  return (...args) => {
+    clearTimeout(timeoutId);
+    timeoutId = setTimeout(() => fn(...args), delay);
+  };
+}
+const debouncedResizeIframe = debounce(resizeIframe);
+
+function setupResizeObserver(iframe) {
+  const iframeContent = iframe.contentWindow.document;
+  const resizeObserver = new ResizeObserver(() => debouncedResizeIframe(iframe));
+
+  resizeObserver.observe(iframeContent.body);
+  resizeIframe(iframe);
+}
+
+if (isIframeLoaded) {
+  setupResizeObserver(iframe);
+} else {
+  iframe.addEventListener('load', () => setupResizeObserver(iframe));
+}
```

Comment:
I ended up adding this line as well, because the iframe content can take a few seconds to populate and kept getting inconsistent resizing behavior, often having to trigger a resize event to get the redraw to take. This seems to properly trigger it without having to touch anything.

`window.addEventListener('load', () => debouncedResizeIframe(iframe));`

Note: There are still some load time issues from the upstream end, right now it can be up to 6 seconds before the iframe fully 'loads'.